partiql-parser/src/scanner.rs

```diff
@@ -399,6 +399,63 @@ mod test {
         };
     }
 
+    #[rstest]
+    #[case::comment_single_keyword(
+        scanner_test_case![
+            "--",
+            "SELECT",
+            " \n ",
+        ]
+    )]
+    #[rstest]
+    #[case::comment_mid_line(
+        scanner_test_case![
+            "SELECT" => keyword("SELECT"),
+            "  ",
+            "FROM" => keyword("FROM"),
+            " -- ",
+            "WHERE",
+            " \n ",
+        ]
+    )]
+    #[rstest]
+    #[case::comment_until_eol(
+        scanner_test_case![
+            " -- ",
+            "CASE",
+            "  ",
+            "IN",
+            "  ",
+            "WHERE",
+            " \n ",
+            "SELECT" => keyword("SELECT"),
+        ]
+    )]
+    #[rstest]
+    #[case::comment_block(
+        scanner_test_case![
+            " /* ",
+            "CASE",
+            "  ",
+            "IN",
+            " */ ",
+            "SELECT" => keyword("SELECT"),
+        ]
+    )]
+    #[rstest]
+    #[case::comment_block_nested(
+        scanner_test_case![
+            "employee" => identifier("employee"),
+            " /* ",
+            "CASE",
+            " /* ",
+            "WHERE",
+            " */ ",
+            "employee",
+            " */ ",
```

Comment:
Would be good to test a couple interesting cases here, some extra newline characters in the mix is a good idea, bit also unbalanced comments as an error case (see `bad_tokens` for a place where the unbalanced comments could go to be tested).